woocommerce/wc-admin
# Allow negative values in charts

Fixes #1959.

Adds the ability to display negative numbers to `<Chart>` component.

### Screenshots
_All negative:_
![image](https://user-images.githubusercontent.com/3616980/55423087-78f36200-557d-11e9-944d-75afb47f3cd4.png)
_Positive and negative values:_
![image](https://user-images.githubusercontent.com/3616980/55423669-a987cb80-557e-11e9-8f60-cd043876b9c2.png)
_Bar charts:_
![image](https://user-images.githubusercontent.com/3616980/55423365-0767e380-557e-11e9-94ab-064a5470b887.png)

### Detailed test instructions:
Currently all our charts only display positive values, so you need to overwrite `chartData` before rendering them.

In https://github.com/woocommerce/woocommerce-admin/blob/master/client/analytics/components/report-chart/index.js#L125, add these lines:
```ES6
for ( let i = 0; i < chartData.length; i++ ) {
	chartData[ i ].primary.value = ( Math.random() - 0.5 ) * 100;
	chartData[ i ].secondary.value = ( Math.random() - 0.5 ) * 100;
}
```
- Go to any report and verify charts look good.
- Switch between line and bar chart.
- Try modifying the code snippet above to force other data combinations:

**Only negative values:**
```ES6
for ( let i = 0; i < chartData.length; i++ ) {
	chartData[ i ].primary.value = Math.random() * -100;
	chartData[ i ].secondary.value = Math.random() * -100;
}
```
  **All 0's:**
```ES6
for ( let i = 0; i < chartData.length; i++ ) {
	chartData[ i ].primary.value = 0;
	chartData[ i ].secondary.value = 0;
}
```
  **Beautiful data going up (used for the screenshots above) :chart_with_upwards_trend: :**
```ES6
for ( let i = 0; i < chartData.length; i++ ) {
	chartData[ i ].primary.value = ( Math.random() - 0.7 + i / 30 ) * 100;
	chartData[ i ].secondary.value = ( Math.random() - 0.7 + i / 30 ) * 100;
}
```
